File: cpp_e9zMZ/cpp/10-tonghop/78.cpp
#include <iostream>
#include <string>
#include <vector>

#include "zvn.h"
using namespace std;

int main() {
    string inputCategory = "D1|Lập trình di động$D2|Lập trình phần mềm$D3|Lập trình nhúng";
    string inputCourse = "1|Dart|D1=2|Golang|D1=3|Laravel|D1=4|TypeScript|D2=5|NodeJS|D2=6|C/C++|D3";

    vector<string> vCategory = splitString(inputCategory, {'|', '$'});
    vector<string> vCourse = splitString(inputCourse, {'|', '='});
    int sizeVCategory = vCategory.size();
    int sizeVCourse = vCourse.size();
    string result = "";

    int index = 1;
    for (int i = 0; i < sizeVCategory; i += 2) {
        string categoryId = vCategory[i];
        string categoryName = vCategory[i + 1];
        string courses = "";
        int count = 0;

        for (int j = 2; j < sizeVCourse; j += 3) {
            if (!categoryId.compare(vCourse[j])) {
                courses += vCourse[j - 1] + ", ";
                count++;
            }
        }

        string record = to_string(index) + ". " + categoryName + " (" + to_string(count) + "): " + courses;
        result += record.substr(0, record.length() - 2) + "\n";
        index++;
    }

    cout << result << endl;

    return 0;
}

File: python_ToSXs/python/10-tonghop/78.py
inputCategory = "D1|Lập trình web$D2|Lập trình phần mềm$D3|Lập trình di động"
inputCourse = "1|HTML CSS|D1=2|PHP|D1=3|Laravel|D1=4|Winform|D2=5|NodeJS|D2=6|React Native|D3"

inputCategory = inputCategory.replace("$", "|")
inputCourse = inputCourse.replace("=", "|")

arrInputCategory = inputCategory.split("|")
arrInputCourse = inputCourse.split("|")

result = ""
index = 1

for i in range(1, len(arrInputCategory), 2):
    categoryID = arrInputCategory[i - 1]
    categoryName = arrInputCategory[i]
    record = f"{str(index)}. {categoryName}"
    count = 0
    listCourse = ""

    for j in range(1, len(arrInputCourse), 3):
        if arrInputCourse[j + 1] == categoryID:
            count += 1
            listCourse += arrInputCourse[j] + ", "

    listCourse = listCourse[0:len(listCourse) - 2]
    record += f"({count}): {listCourse}"
    result += f"{record} \n"
    index += 1

print(result)
